### PR TITLE
pdgui_vmess() fixes and improvements

### PR DESCRIPTION
**Pull request by Spacechild1**
The first commit (10a2e9d756e0179b1c94c066b8dfa479d5a75a1e) fixes a critical crasher bug (https://github.com/pure-data/pure-data/issues/1751).

The remaining 3 commits are mere improvements.

### CODE REVIEW

File: src/s_main.c [hunk 1]
```diff
@@ -1518,7 +1518,7 @@ static void namelist2gui(const char*name, t_namelist*namelist)
     }
     pdgui_vmess("set", "rS",
               name,
-              i, names);
+              (int)i, names);
```
Comment (umlaeute) [inline src/s_main.c:1521]: good catch.

but i think the fix is wrong. instead, `i` should just be made an `int` (which I did with a8a245d3752af4026a17bb2c1ecb5d7c1a18606e)

**Review by umlaeute — COMMENTED** [review]
good catch for the `va_list` issue.

i've merged that into `develop` (with my small amendment of a450c160726b24db79205a2ba3c6dd11fffd518d

**Comment by umlaeute** [comment]
> good catch for the `va_list` issue.

hmm. actually it now crashes on Linux :-(

the issue seems to be that on Linux and macOS, `va_list` is implemented as an array (rather than just a pointer).
see https://stackoverflow.com/a/57805088/1169096

hopefully i've submitted a fix with dbcaf3c6b40fe9866ba0d663dd2f91c10d0fab46 that works on all platforms.
this time tested on:
- Linux/amd64
- macOS/amd64
- Wine/amd64
- Wine/i386

**Comment by Spacechild1** [comment]
That's wild! I have read the explanation in https://stackoverflow.com/a/57805088/1169096 several times now and still don't really understand it :-/ How is taking the address of a local variable (`&ap`) different from taking the address of a function argument (`&ap_`)? But as long as it works... :-)

**Comment by Spacechild1** [comment]
Anyway, I can confirm that https://github.com/pure-data/pure-data/commit/dbcaf3c6b40fe9866ba0d663dd2f91c10d0fab46 works fine also on Windows.

**Comment by Spacechild1** [comment]
> How is taking the address of a local variable (&ap) different from taking the address of a function argument (&ap_)? 

Now I get it! An `va_list` array argument actually decays to a pointer to the first argument, so the `&` operator yields a pointer to a pointer to the first argument. The `va_list` array on the stack does not decay, so `&` just refers to the array.

C++ would have caught this at compile time (incompatible pointer types)... *sigh*

**Pull request merged**